packages/gatsby-transformer-sharp/src/customize-schema.js

```diff
@@ -546,3 +538,35 @@ module.exports = ({
     },
   }
 }
+
+module.exports = ({
+  actions,
+  schema,
+  pathPrefix,
+  getNodeAndSavePathDependency,
+  reporter,
+  cache,
+}) => {
+  const { createTypes } = actions
+
+  const imageSharpType = schema.buildObjectType({
+    name: `ImageSharp`,
+    fields: createFields({
+      pathPrefix,
+      getNodeAndSavePathDependency,
+      reporter,
+      cache,
+    }),
+    interfaces: [`Node`],
+    extensions: {
+      infer: true,
```

Comment:
Let's actually try to get it to `infer: false`. In addition to fields that we add in `createFields`, could you look at sourceNodes code for this plugin and see what fields are created to be inferred?

Reply:
There was a discussion that we should keep it for BC but also add the `deprecated` version of `infer` which would warn when new node fields were added without corresponding schema extension.

Reply:
Ok, we don't have to block this based on this. Could you add all the existing fields we add in the plugin and then we can merge it with infer: true, with a note that we'll deprecate it when deprecated infer lands.

Reply:
@freiksenet I compared schemas generated with `infer: false` and `infer: true` and they are already identical, so I assume this plugin already describes all the fields

Reply:
OK, then I was confused about this :) Sorry about it. This is good to be merged.